### PR TITLE
[fix](schema-change) Fix schema-change from non-null to null

### PR DESCRIPTION
**Pull request by Lchangliang**
https://github.com/apache/doris/pull/32913

### CODE REVIEW

**Comment by Lchangliang** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by Lchangliang** [comment]
run buildall

File: be/src/olap/schema_change.cpp [hunk 1]
```diff
@@ -389,11 +389,22 @@ Status BlockChanger::_check_cast_valid(vectorized::ColumnPtr ref_column,
...
-                            .data();
+                            ->get_null_map_column();
+            const auto& nested_column =
+                    vectorized::check_and_get_column<vectorized::ColumnNullable>(new_column)
+                            ->get_nested_column();
+            const auto* new_null_map = null_map_column.get_data().data();
+
+            if (null_map_column.size() != new_column->size() ||
+                nested_column.size() != new_column->size()) {
+                DCHECK(false);
```
Comment (gavinchou) [inline be/src/olap/schema_change.cpp:402]: Print more details like size and some ids or names

**Comment by Lchangliang** [comment]
run buildall

**Pull request merged**